### PR TITLE
Added method for cross bi-directional trust attack

### PR DESCRIPTION
**Pull request by martinsohn**
I have added a trust attack technique with GoldenGMSA to the code and readme. I also fixed a few things in the readme.

Please see attached PDF for details and demonstrations:
[GoldenGMSA trust attack.pdf](https://github.com/Semperis/GoldenGMSA/files/8186241/GoldenGMSA.trust.attack.pdf)

### CODE REVIEW

File: GoldenGMSA/Program.cs [hunk 1]
```diff
@@ -116,13 +134,30 @@ static void ProcessComputePwdOptions(ComputePwdOptions options)
                 if (options.Sid == null)
                     throw new ArgumentNullException(nameof(options.Sid));
```
Comment (GuyTe) [inline GoldenGMSA/Program.cs:136]: This cancels the offline mode and tries to talk to AD even when all the required arguments have been passed on the command line (sid/KdsRootKeyBase64/ManagedPwdIdBase64). Offline mode does not require the domain/forest names.

**Review by GuyTe — CHANGES_REQUESTED** [review]

**Review by Quazaka — APPROVED** [review]

**Review by martinsohn — COMMENTED** [review]
Hi Guy, thank you for the fix, can you merge into master?

**Comment by GuyTe** [comment]
> Hi Guy, thank you for the fix, can you merge into master?

I don't see how you have addresses the point I
 have raised. The function should be able to operate in "offline" mode when all the required arguments have been provided (i.e.: both KdsRootKeyBase64 and ManagedPwdIdBase64 have been passed as arguments). In this scenario no AD calls are required and rest of the arguments should be ignored

**Comment by MartinSohn** [comment]
I gotcha @GuyTe, please have a look at the newest commit.
Here I query both domains for info and compute online:
image

Here I compute offline by giving previously returned values:
image

**Review by GuyTe — APPROVED** [review]

**Comment by GuyTe** [comment]
Thanks @MartinSohn for the additions and the cleanup. Much appreciated

**Comment by jsdhasfedssad** [comment]
Thank you for PR which I will surely use in future. Is there any way to convert the base64 encoded password to an NT hash? This would open up for a lot more tools.

I have decoded the base64 encoded password using different methods but they all end up in an unusable state.

**Pull request merged**